modules/standard/Random.chpl

```diff
@@ -359,7 +363,7 @@ module Random {
       // Return 1 sample
       var randNum = stream.getNext(resultType=real);
       var (found, idx) = Search.binarySearch(cumulativeArr, randNum);
-      return A[idx];
+      return (idx-1) * stride + low;
```

Comment:
I wonder if you could use [`range.orderToIndex`](https://chapel-lang.org/docs/master/builtins/ChapelRange.html#ChapelRange.range.orderToIndex) here?

Reply:
@ben-albrecht before that I have a doubt. After shifting to 0 based indexing can `idx` be zero now? I am not much sure about that.

Reply:
> I wonder if you could use [`range.orderToIndex`](https://chapel-lang.org/docs/master/builtins/ChapelRange.html#ChapelRange.range.orderToIndex) here?

Yes, depending on start value of `idx` we can do something like 

```chapel
X.dim(0).orderToIndex(idxl);
```
or 

```chapel
X.dim(0).orderToIndex(randVal);
```